### PR TITLE
Beta100 read me update

### PR DESCRIPTION
**Pull request by jakobuid**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,223 +33,230 @@ implementation("com.github.WalletConnect:WalletConnectKotlinV2:release_version")
...
 
 The wallet client will always be responsible for exposing accounts (CAPI10 compatible) to a Dapp and therefore is also in charge of signing.
-To initialize the WalletConnect client, create a `WalletConnect.Params.Init` object in the Android Application class. The Init object will
-need the application class, the Project ID, and the wallet's AppMetaData. The `WalletConnect.Params.Init` object will then be passed to
-the `WalletConnectClient` initialize function. `WalletConnect.Params.Init` also allows for custom URLs by passing URL string into
-the `hostName`
-property. Above, there are two example on how to create the initalizing parameters.
+To initialize the Sign client, create a `Sign.Params.Init` object in the Android Application class. The Init object will need the
+application class, the Project ID, and the apps's AppMetaData. The `Sign.Params.Init` object will then be passed to the `SignClient`
+initialize function. `Sign.Params.Init` also allows for custom URLs by passing URL string into the `hostName`property.
```
Comment (Elyniss) [inline README.md:72]:
```suggestion
initialize function. `Sign.Params.Init` also allows for custom URLs by passing URL string into the `hostName` property.
```

File: README.md [hunk 2]
```diff
@@ -32,223 +33,230 @@ implementation("com.github.WalletConnect:WalletConnectKotlinV2:release_version")
...
+
+We allow developers to choose between the `Sign.ConnectionType.MANUAL`and`Sign.ConnectionType.AUTOMATIC`connection type. The default
```
Comment (Elyniss) [inline README.md:74]:
```suggestion
We allow developers to choose between the `Sign.ConnectionType.MANUAL` and `Sign.ConnectionType.AUTOMATIC`connection type. The default
```

File: README.md [hunk 3]
```diff
@@ -32,223 +33,230 @@ implementation("com.github.WalletConnect:WalletConnectKotlinV2:release_version")
...
 
-    override fun onSessionUpdateAccountsResponse(response: WalletConnect.Model.SessionUpdateAccountsResponse) {
-         // Triggered when wallet receives the update accounts response from Dapp
+    override fun onSessionSettleResponse(settleSessionResponse: Sign.Model.SettledSessionResponse) {
+        // Triggered when wallet receives the session settlement response from Dapp
     }
 
-    override fun onSessionUpdateMethodsResponse(response: WalletConnect.Model.SessionUpdateMethodsResponse) {
-         // Triggered when wallet receives the update methods response from Dapp
+    fun onSessionUpdateResponse(sessionUpdateResponse: WalletConnect.Model.SessionUpdateResponse) {
```
Comment (Elyniss) [inline README.md:105]:
```suggestion
    fun onSessionUpdateResponse(sessionUpdateResponse: Sign.Model.SessionUpdateResponse) {
```

File: README.md [hunk 4]
```diff
@@ -32,223 +33,230 @@ implementation("com.github.WalletConnect:WalletConnectKotlinV2:release_version")
...
+        // Triggered when wallet receives the session update response from Dapp
     }
 
-    fun onSessionUpdateEventsResponse(response: WalletConnect.Model.SessionUpdateEventsResponse) {
-         // Triggered when wallet receives the update events response from Dapp
+    fun onConnectionStateChange(state: WalletConnect.Model.ConnectionState) {
```
Comment (Elyniss) [inline README.md:109]:
```suggestion
    fun onConnectionStateChange(state: Sign.Model.ConnectionState) {
```

File: README.md [hunk 5]
```diff
@@ -32,223 +33,230 @@ implementation("com.github.WalletConnect:WalletConnectKotlinV2:release_version")
...
 
 ### **Session Approval**
-NOTE: addresses provided in `accounts` array should follow [CAPI10](https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-10.md) semantics.
+
+NOTE: addresses provided in `accounts` array should follow [CAPI10](https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-10.md)
+semantics.
+
 ```kotlin
 val proposerPublicKey: String = /*Proposer publicKey from SessionProposal object*/
+val namespace = /*Namespace identifier, see for reference: https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-2.md#syntax*/
```
Comment (Elyniss) [inline README.md:139]:
```suggestion
val namespace: String = /*Namespace identifier, see for reference: https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-2.md#syntax*/
```

File: README.md [hunk 6]
```diff
@@ -32,223 +33,230 @@ implementation("com.github.WalletConnect:WalletConnectKotlinV2:release_version")
...
 &nbsp;
 
 ### **Session Rejection**
+
 ```kotlin
 val proposerPublicKey: String = /*Proposer publicKey from SessionProposal object*/
 val rejectionReason: String = /*The reason for rejecting the Session Proposal*/
 val rejectionCode: String = /*The code for rejecting the Session Proposal*/
-For reference use CAIP-25: https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-25.md
+For reference use CAIP -25: https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-25.md
```
Comment (Elyniss) [inline README.md:159]:
```suggestion
For reference use CAIP-25: https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-25.md
```

File: README.md [hunk 7]
```diff
@@ -32,223 +33,230 @@ implementation("com.github.WalletConnect:WalletConnectKotlinV2:release_version")
...
 
 ### **Session Disconnect**
+
 ```kotlin
 val disconnectionReason: String = /*The reason for disconnecting the Session*/
 val disconnectionCode: String = /*The code for for disconnecting the Session*/
 val sessionTopic: String = /*Topic from the Session*/
-For reference use CAIP-25: https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-25.md
-val disconnectParams = WalletConnect.Params.Disconnect(sessionTopic, disconnectionReason, disconnectionCode)
+For reference use CAIP -25: https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-25.md
```
Comment (Elyniss) [inline README.md:176]:
```suggestion
For reference use CAIP-25: https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-25.md
```

File: README.md [hunk 8]
```diff
@@ -32,223 +33,230 @@ implementation("com.github.WalletConnect:WalletConnectKotlinV2:release_version")
...
+val jsonRpcResponse: Sign.Model.JsonRpcResponse.JsonRpcResult = /*Settled Session Request ID along with request data*/
+val result = Sign.Params.Response(sessionTopic = sessionTopic, jsonRpcResponse = jsonRpcResponse)
 
-WalletConnectClient.respond(result) { error -> /*optional callback for error while responding session request*/ }
+SignClient.respond(result) { error -> /*callback for error while responding session request*/ }
 ```
-To reject a JSON-RPC method that was sent from a Dapps for a session, submit a `WalletConnect.Params.Response` with the settled session's topic and request ID along with the rejection data to the `WalletConnectClient.respond` function.
 
-&nbsp;
+To respond to JSON-RPC methods that were sent from Dapps for a session, submit a `Sign.Params.Response` with the session's topic and request
```
Comment (Elyniss) [inline README.md:197]:
```suggestion
To respond to JSON-RPC method that were sent from Dapps for a session, submit a `Sign.Params.Response` with the session's topic and request
```

File: README.md [hunk 9]
```diff
@@ -32,223 +33,230 @@ implementation("com.github.WalletConnect:WalletConnectKotlinV2:release_version")
...
-&nbsp;
+NOTE: addresses provided in `accounts` array should follow [CAPI10](https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-10.md)
+semantics.
 
-### **Session Update Events**
 ```kotlin
 val sessionTopic: String = /*Topic of Session*/
-val events: List<String> = /*List of methods to update*/
-val updateParams = WalletConnect.Params.UpdateEvents(sessionTopic = sessionTopic, events = events)
+val namespace = /*Namespace identifier, see for reference: https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-2.md#syntax*/
```
Comment (Elyniss) [inline README.md:222]:
```suggestion
val namespace: String = /*Namespace identifier, see for reference: https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-2.md#syntax*/
```

File: README.md [hunk 10]
```diff
@@ -258,108 +266,108 @@ val listener = object : WalletConnect.Listeners.SessionPing {
...
-The WalletConnectClient needs a `WalletConnectClient.DappDelegate` passed to it for it to be able to expose asynchronously updates sent from the
-Wallet.
+The SignClient needs a `SignClient.DappDelegate` passed to it for it to be able to expose asynchronously updates sent from the Wallet.
 
 &nbsp;
 
 ### **Connect**
 
 ```kotlin
+val namespace = /*Namespace identifier, see for reference: https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-2.md#syntax*/
```
Comment (Elyniss) [inline README.md:324]:
```suggestion
val namespace: String = /*Namespace identifier, see for reference: https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-2.md#syntax*/
```

File: README.md [hunk 11]
```diff
@@ -258,108 +266,108 @@ val listener = object : WalletConnect.Listeners.SessionPing {
...
     }
 
-    override fun onSessionRejected(rejectedSession: WalletConnect.Model.RejectedSession) {
+    override fun onSessionRejected(rejectedSession: Sign.Model.RejectedSession) {
         // Triggered when Dapp receives the session rejection from wallet
     }
 
-    override fun onSessionUpdateAccounts(updatedSession: WalletConnect.Model.UpdatedSessionAccounts) {
-        // Triggered when Dapp receives the session update accounts from wallet
+    fun onSessionUpdate(updatedSession: WalletConnect.Model.UpdatedSession) {
```
Comment (Elyniss) [inline README.md:290]:
```suggestion
    fun onSessionUpdate(updatedSession: Sign.Model.UpdatedSession) {
```

File: README.md [hunk 12]
```diff
@@ -258,108 +266,108 @@ val listener = object : WalletConnect.Listeners.SessionPing {
...
+        // Triggered when Dapp receives the session update from wallet
     }
 
-    fun onSessionUpdateMethods(updatedSession: WalletConnect.Model.UpdatedSessionMethods) {
-        // Triggered when Dapp receives the session update methods from wallet
+    fun onSessionExtend(session: WalletConnect.Model.Session) {
```
Comment (Elyniss) [inline README.md:294]:
```suggestion
    fun onSessionExtend(session: Sign.Model.Session) {
```

File: README.md [hunk 13]
```diff
@@ -258,108 +266,108 @@ val listener = object : WalletConnect.Listeners.SessionPing {
...
 
-    fun onSessionDelete(deletedSession: WalletConnect.Model.DeletedSession) {
-           // Triggered when Dapp receives the session delete from wallet
+    fun onSessionRequestResponse(response: Sign.Model.SessionRequestResponse) {
+        // Triggered when Dapp receives the session request response from wallet
     }
 
-    fun onSessionRequestResponse(response: WalletConnect.Model.SessionRequestResponse) {
-           // Triggered when Dapp receives the session request response from wallet
+    fun onConnectionStateChange(state: WalletConnect.Model.ConnectionState) {
```
Comment (Elyniss) [inline README.md:310]:
```suggestion
    fun onConnectionStateChange(state: Sign.Model.ConnectionState) {
```

File: README.md [hunk 14]
```diff
@@ -258,108 +266,108 @@ val listener = object : WalletConnect.Listeners.SessionPing {
...
 
 ```kotlin
+val namespace = /*Namespace identifier, see for reference: https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-2.md#syntax*/
 val chains: List<String> = /*List of chains that wallet will be requested for*/
 val methods: List<String> = /*List of methods that wallet will be requested for*/
 val events: List<String> = /*List of events that wallet will be requested for*/
+val namespaces: Map<String, Sign.Model.Namespaces.Proposal> = mapOf(namespace, Sign.Model.Namespaces.Proposal(accounts, methods, events))
 val pairingTopic: String? =  /* Optional parameter, use it when the pairing between peers is already established*/
-val connectParams = WalletConnect.Params.Connect(chains, methods, events, pairingTopic)
+val connectParams = WalletConnect.Params.Connect(namespaces, pairingTopic)
```
Comment (Elyniss) [inline README.md:330]:
```suggestion
val connectParams = Sign.Params.Connect(namespaces, pairingTopic)
```

**Review by Elyniss — CHANGES_REQUESTED** [review]

**Review by Elyniss — APPROVED** [review]

File: README.md [hunk 15]
```diff
@@ -32,223 +33,230 @@ implementation("com.github.WalletConnect:WalletConnectKotlinV2:release_version")
...
+        //Triggered whenever the connection state is changed
     }
 }
-WalletConnectClient.setWalletDelegate(walletDelegate)
+SignClient.setWalletDelegate(walletDelegate)
 ```
 
-The WalletConnectClient needs a `WalletConnectClient.WalletDelegate` passed to it for it to be able to expose asynchronously updates sent from
-the Dapp.
+The SignClient needs a `SignClient.WalletDelegate` passed to it for it to be able to expose asynchronously updates sent from the Dapp.
```
Comment (TalhaAli00) [inline README.md:116]: change `asynchronously` to `asynchronous`

**Pull request merged**